src/servant/tt_002/3678.c

```diff
@@ -1029,7 +1034,210 @@ void func_us_80176B6C(Entity* self) {
+                self->step = 5;
+            } else if (s_FaerieStats.level < 0x5A && rand() % 8) {
+                self->step = 5;
+            } else {
+                self->ext.faerie.unkB4 = 0;
+                if (PLAYER.step_s == 4) {
+                    self->ext.faerie.frameCounter = D_us_80172D2A;
+                } else {
+                    self->ext.faerie.frameCounter = D_us_80172D28;
+                }
+                self->step++;
+            }
+#endif
+
+
+        }
+        break;
+    case 2:
+        --self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter < 0) {
+            self->ext.faerie.frameCounter = 0;
+            if (g_api.func_800F27F4(0)) {
+                func_us_80173994(self, 0x19);
+                self->ext.faerie.unkB4 = 1;
+                self->step++;
+            }
+        }
+        break;
+    case 3:
+        ++self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter > 0x420) {
+            func_us_80173994(self, 0x1A);
+            self->ext.faerie.frameCounter = 0;
+            self->step++;
+        }
+        break;
+    case 4:
+        ++self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter > 0x15E0) {
+            func_us_80173994(self, 0x19);
+            self->ext.faerie.frameCounter = 0;
+            self->step++;
+        }
+        break;
+    case 5:
+        ++self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter > 0x500) {
+            func_us_80173994(self, 0x18);
+            self->ext.faerie.frameCounter = 0;
+            self->ext.faerie.unkB4 = 0;
+            self->step++;
+        }
+        break;
+    case 6:
+        self->ext.faerie.frameCounter = (rand() % 0x800) + 0x400;
+        self->step++;
+        /* fallthrough */
+    case 7:
+        --self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter < 0) {
+            self->ext.faerie.frameCounter = (rand() % 0x80) + 0x80;
+            func_us_80173994(self, 0x19);
+            self->step++;
+        }
+        break;
+    case 8:
+        --self->ext.faerie.frameCounter;
+        if (self->ext.faerie.frameCounter < 0) {
+            func_us_80173994(self, 0x18);
+            self->step = 6;
+        }
+        break;
+    case 9:
+        if (self->ext.faerie.unkB4) {
+            g_api.func_800F27F4(1);
+        }
+        self->flags &= ~FLAG_POS_PLAYER_LOCKED;
+        self->flags |= FLAG_POS_CAMERA_LOCKED;
+        func_us_80173994(self, 0x1B);
+        self->velocityX = self->facingLeft ? FIX(-0.25) : FIX(0.25);
+        self->velocityY = FIX(1);
+        for (rnd = rand() % 0x100, i = 0; true; i++) {
+            if(rnd <= *(D_us_80172C04 + (i * 2)))
+            {
+#ifdef VERSION_PSP
+                g_api.PlaySfx(*(D_us_80172C08 + (i * 2 + 1)));
+#else
+                g_api.PlaySfx(*(D_us_80172C08 + (i * 2)));
```

Comment:
I'm pretty sure this whole ifdef block can be replaced with
g_api.PlaySfx(D_us_80172C04[i * 2 + 1]);

Then you can remove D_us_80172C08[]

I think it should be the same for both PSX and PSP